pkg/node/kubelet.go

```diff
@@ -16,26 +16,61 @@ limitations under the License.
 package node
 
 import (
+	"context"
 	"path/filepath"
 	"strconv"
 
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 
 	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util"
 
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	cliflag "k8s.io/component-base/cli/flag"
 	kubeproxy "k8s.io/kubernetes/cmd/kube-proxy/app"
 	kubelet "k8s.io/kubernetes/cmd/kubelet/app"
+
+	kubeletoptions "k8s.io/kubernetes/cmd/kubelet/app/options"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+)
+
+const (
+	// Kubelet component name
+	componentKubelet = "kubelet"
 )
 
-func StartKubelet(cfg *config.MicroshiftConfig) error {
-	command := kubelet.NewKubeletCommand()
+type KubeletServer struct {
+	kubeletflags *kubeletoptions.KubeletFlags
+	kubeconfig   *kubeletconfig.KubeletConfiguration
+}
+
+func NewKubeletServer(cfg *config.MicroshiftConfig) *KubeletServer {
+	s := &KubeletServer{}
+	s.configure(cfg)
+	return s
+}
+
+func (s *KubeletServer) Name() string           { return componentKubelet }
+func (s *KubeletServer) Dependencies() []string { return []string{"kube-apiserver"} }
+
+func (s *KubeletServer) configure(cfg *config.MicroshiftConfig) error {
+
+	//create KubeletConfiguration file at cfg.DataDir + "/resources/kubelet/config/config.yaml
+	if err := config.KubeletConfig(cfg); err != nil {
+		logrus.Infof("Failed to create a new kubelet configuration: %v", err)
+		return err
+	}
+	// Prepare commandline args
 	args := []string{
 		"--config=" + cfg.DataDir + "/resources/kubelet/config/config.yaml",
 		"--bootstrap-kubeconfig=" + cfg.DataDir + "/resources/kubelet/kubeconfig",
 		"--kubeconfig=" + cfg.DataDir + "/resources/kubelet/kubeconfig",
 		"--container-runtime=remote",
 		"--container-runtime-endpoint=/var/run/crio/crio.sock",
 		"--runtime-cgroups=/system.slice/crio.service",
+		"--cgroup-driver=systemd",
```

Comment:
this is already set [here](https://github.com/redhat-et/microshift/blob/main/pkg/config/kubelet.go#L37)